### ISSUE TITLE
Alignment should account for long instruction names

### ISSUE THREAD

**Issue by eyalroz** [compiler-explorer/compiler-explorer#863]
Usually, the assembly output is aligned using spaces on the first argument, e.g.:
```
        mov     rcx, qword ptr [r8 + rax + 32]
        imul    rcx, qword ptr [rdx + rax + 32]
        add     rcx, qword ptr [rsi + rax + 32]

```
However, when the compilers generate instructions with very long names, the alignment is not pushed rightwards to accommodate for these, and we get, for example:
```
        vmovsd  xmm0, qword ptr [rdx + rax + 32] # xmm0 = mem[0],zero
        vmovsd  xmm1, qword ptr [rcx + rax + 32] # xmm1 = mem[0],zero
        vfmadd213sd     xmm1, xmm0, qword ptr [rsi + rax + 32]
        vmovsd  qword ptr [rdi + rax + 32], xmm1
```
which is kind of ugly and less readable. Now, assuming the user has not enabled `\s+`, I believe the arguments should be aligned by ensuring there's enough space.

**Comment by mattgodbolt** [comment]
I agree it'd be lovely to fix this. GCC and clang only put a single tab between instructions and their operands, which means as soon as an instruction is 8 or more characters long, the indent doesn't work. We can of course parse the instructions, and re-flow them, but then *all* instructions would be super-indented to account for a single very long instructions. I think this is why the compilers stick to a single tab character, instead of using two or more. I guess we could treat tabs as being (say) 12 characters long which I _think_ covers all current x86 instructions...

**Comment by eyalroz** [comment]
The 12-char tab sound good to me (but won't that affect the leading space on instruction lines?)

You could also theoretically make the tab length configurable.

**Comment by mattgodbolt** [comment]
Happy to accept patches for this. If it's an option we probably need to fix up our UI to make space for more options as there's precious little room as it is.

**Comment by eyalroz** [comment]
@mattgodbolt : It's unlikely I will be getting into your code anytime soon, unfortunately...

About the UI fixup - it's a separate issue of course, but I will say that the common solution is to have a "menu with more stuff that doesn't fit" button on the right or leftmost side, marked by vertical ellipsis, or a right guillemet (>>), or three horizontal lines like in browsers. But you probably know that :-)

**Comment by mattgodbolt** [comment]
@eyalroz absolutely. that's my thought too re: UX, it's just a question of implementing it.

**Comment by mattgodbolt** [comment]
I think I would actually quite like this... but you can assign it to me for a snowy/rainy day... and/or "help wanted/good first issues" as depending on having a smart person come along there might be a neat trick!

**Comment by jinder1s** [comment]
Hi all, 
I'm thinking of looking into this. I'm wondering, do you folks have any example that produced compiler output with long names. 
I'm very new to this tool, so still learning my way around.

**Comment by RubenRBS** [comment]
Thanks for taking the time to check this issue, let us know if you find any other problem where we can help :)

**Comment by avg-itmo-enjoyer** [comment]
@mattgodbolt I've been searching for an issue with "good first issue" tag and saw this one. Checked - it's still reproducible. May I take it and try to implement (at least test few options discussed above)?

**Comment by mattgodbolt** [comment]
Please feel free: usually one doesn't need to ask: just let us know if we can help and of course once it's ready send a PR. Thank you
